### ISSUE TITLE
[TM1701] System crashes when running memory intense operations.

### ISSUE THREAD

**Issue by MrFoxPro** [daliansky/XiaoMi-Pro-Hackintosh#740]
**State the model**
Xiaomi Mi Notebook Pro i7-8550U, 16Gb, MX150
~MacOS 13.5.2~ (Same on 12, 13, 14)
MacOS installed on 512GB SSD with `Silicon Motion SM2263XT` controller in 0 slot alongside with Windows and other partitions, and second SSD with Linux has `Silicon Power` controller.

**State the version of the EFI**
Tried 1.7.9 and [beta-8c15a8f](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/suites/15963592305/artifacts/911399912)

**Describe the bug**
System crashes when running memory intense operations: unpacking Xcode xip, running IOS simulator. I tried OpenCore 1.7.0, OpenCore beta, and Clover from mentioned before beta.

**NOTE**: I have patched EC604 BIOS (patch for fans temp by Cyb). 

I also removed AirportItlwm_* for previous MacOS versions from `Kexts` folder and removed it from `config.plist`. But before this it was crashing too.

**System log**
<details>
  <summary>timed-out on cpu 4, NMI-ing</summary>
  
```
mp_kdp_enter() timed-out on cpu 4, NMI-ing
mp_kdp_enter() NMI pending on cpus: 0 1 2 5 7
mp_kdp_enter() timed-out during locked wait after NMI;expected 8 acks but 
received 4 after 7811243 loops in 996000000 ticks
panic(cpu 4 caller 0xffffff80019b6e07): Double fault at 
0xffffff8001812a0a, registers:
CR0: 0x000000008001003b, CR2: 0xffffff801635cff8, CR3: 0x00000003323dc170, 
CR4: 0x00000000003626e0
RAX: 0x000000008001003b, RBX: 0x00000000000000de, RCX: 0xfffffff3f9507fb0, 
RDX: 0x0000000000000000
RSP: 0xffffff801635d000, RBP: 0x0000000000000000, RSI: 0x000000000000002b, 
RDI: 0x0000000000000000
R8:  0x0000000000000000, R9:  0x0000000000000000, R10: 0x0000000000000000, 
R11: 0x0000000000000000
R12: 0x0000000000000000, R13: 0x0000000000000000, R14: 0x0000000000000000, 
R15: 0xffffffa0782c1ac0
RFL: 0x0000000000010082, RIP: 0xffffff8001812a0a, CS:  0x0000000000000008, 
SS:  0x0000000000000000
Error code: 0x0000000000000000
 @trap_native.c:168
Panicked task 0xffffffaa0d5649d8: 31 threads: pid 561: SpringBoard
Backtrace (CPU 4), panicked thread: 0xffffffaa0df61b30, Frame : Return 
Address
0xffffff800150e270 : 0xffffff800187207d 
0xffffff800150e2c0 : 0xffffff80019c6164 
0xffffff800150e300 : 0xffffff80019b5c77 
0xffffff800150e350 : 0xffffff8001812951 
0xffffff800150e370 : 0xffffff800187235d 
0xffffff800150e460 : 0xffffff8001871a07 
0xffffff800150e4c0 : 0xffffff8001fdb40b 
0xffffff800150e5b0 : 0xffffff80019b6e07 
0xffffff800150e6c0 : 0xffffff8001fe4e38 
0xffffff800150e6d0 : 0xffffff8001812e4f 

Process name corresponding to current thread (0xffffffaa0df61b30): 
SpringBoard
Boot args: darkwake=0 ps2kbdonly=1

Mac OS version:
22G91

Kernel version:
Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; 
root:xnu-8796.141.3~6/RELEASE_X86_64
Kernel UUID: 8295C77B-2AAB-3471-81B5-438F59B1A36D
roots installed: 0
KernelCache slide: 0x0000000001400000
KernelCache base:  0xffffff8001600000
Kernel slide:      0x00000000014dc000
Kernel text base:  0xffffff80016dc000
__HIB  text base: 0xffffff8001500000
System model name: MacBookPro15,4 (Mac-53FDB3D8DB8CA971)
System shutdown begun: NO
Panic diags file available: YES (0x0)
Hibernation exit count: 0

System uptime in nanoseconds: 53229004595
Last Sleep:           absolute           base_tsc          base_nano
  Uptime  : 0x0000000c64b22a68
  Sleep   : 0x0000000000000000 0x0000000000000000 0x0000000000000000
  Wake    : 0x0000000000000000 0x00000004d6c63cb6 0x0000000000000000
Compressor Info: 0% of compressed pages limit (OK) and 0% of segments 
limit (OK) with 0 swapfiles and OK swap space
Zone info:
  Zone map: 0xffffff9a0da90000 - 0xffffffba0da90000
  . PGZ   : 0xffffff9a0da90000 - 0xffffff9a0fa91000
  . VM    : 0xffffff9a0fa91000 - 0xffffff9edc290000
  . RO    : 0xffffff9edc290000 - 0xffffffa075a90000
  . GEN0  : 0xffffffa075a90000 - 0xffffffa542290000
  . GEN1  : 0xffffffa542290000 - 0xffffffaa0ea90000
  . GEN2  : 0xffffffaa0ea90000 - 0xffffffaedb290000
  . GEN3  : 0xffffffaedb290000 - 0xffffffb3a7a90000
  . DATA  : 0xffffffb3a7a90000 - 0xffffffba0da90000
  Metadata: 0xfffffff5f9383000 - 0xfffffff619383000
  Bitmaps : 0xfffffff619383000 - 0xfffffff61c383000
  Extra   : 0 - 0
```
</details>

<details>
<summary>Possible memory corruption</summary>

```
panic(cpu 4 caller 0xffffff801bb95d23): Possible memory corruption: 
pmap_pv_remove(0xffffff801c72af00, 0xffffff803053a000, 0x11398b, 
0x1000011398ba32, 0xffffffce9cf97674, 0xffffffce9cf97610): pv not on hash, 
head: 0xffffff9c0cffc200, 0x12a209000, priors: 1 @pmap_internal.h:979
Panicked task 0xffffff9c0be80bf8: 7 threads: pid 90: configd
Backtrace (CPU 4), panicked thread: 0xffffff9740b84b30, Frame : Return 
Address
0xffffffce9cf97230 : 0xffffff801ba7207d 
0xffffffce9cf97280 : 0xffffff801bbc6164 
0xffffffce9cf972c0 : 0xffffff801bbb5c77 
0xffffffce9cf97310 : 0xffffff801ba12951 
0xffffffce9cf97330 : 0xffffff801ba7235d 
0xffffffce9cf97420 : 0xffffff801ba71a07 
0xffffffce9cf97480 : 0xffffff801c1db40b 
0xffffffce9cf97570 : 0xffffff801bb95d23 
0xffffffce9cf976a0 : 0xffffff801bb13b17 
0xffffffce9cf97720 : 0xffffff801bb126a8 
0xffffffce9cf977a0 : 0xffffff801bb111be 
0xffffffce9cf97a20 : 0xffffff801bbb5bb6 
0xffffffce9cf97a90 : 0xffffff801ba12951 
0xffffffce9cf97ab0 : 0xffffff801bf5e386 
0xffffffce9cf97c10 : 0xffffff801bf88672 
0xffffffce9cf97c40 : 0xffffff801bc8eda3 
0xffffffce9cf97cd0 : 0xffffff801bc8072f 
0xffffffce9cf97d40 : 0xffffff801bc75204 
0xffffffce9cf97d90 : 0xffffff801bc67d1c 
0xffffffce9cf97e00 : 0xffffff801bf63dd9 
0xffffffce9cf97e50 : 0xffffff801bf641a7 
0xffffffce9cf97ec0 : 0xffffff801bf644b1 
0xffffffce9cf97f40 : 0xffffff801c05adc0 
0xffffffce9cf97fa0 : 0xffffff801ba12d96 

Process name corresponding to current thread (0xffffff9740b84b30): configd
Boot args: darkwake=0 ps2kbdonly=1

Mac OS version:
22G91

Kernel version:
Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; 
root:xnu-8796.141.3~6/RELEASE_X86_64
Kernel UUID: 8295C77B-2AAB-3471-81B5-438F59B1A36D
roots installed: 0
KernelCache slide: 0x000000001b600000
KernelCache base:  0xffffff801b800000
Kernel slide:      0x000000001b6dc000
Kernel text base:  0xffffff801b8dc000
__HIB  text base: 0xffffff801b700000
System model name: MacBookPro15,4 (Mac-53FDB3D8DB8CA971)
System shutdown begun: NO
Panic diags file available: YES (0x0)
Hibernation exit count: 0

System uptime in nanoseconds: 1732196694775
Last Sleep:           absolute           base_tsc          base_nano
  Uptime  : 0x000001934ef7ff67
  Sleep   : 0x0000000000000000 0x0000000000000000 0x0000000000000000
  Wake    : 0x0000000000000000 0x0000000a7cb22608 0x0000000000000000
Compressor Info: 0% of compressed pages limit (OK) and 0% of segments 
limit (OK) with 0 swapfiles and OK swap space
Zone info:
  Zone map: 0xffffff90d7dd6000 - 0xffffffb0d7dd6000
  . PGZ   : 0xffffff90d7dd6000 - 0xffffff90d9dd7000
  . VM    : 0xffffff90d9dd7000 - 0xffffff95a65d6000
  . RO    : 0xffffff95a65d6000 - 0xffffff973fdd6000
  . GEN0  : 0xffffff973fdd6000 - 0xffffff9c0c5d6000
  . GEN1  : 0xffffff9c0c5d6000 - 0xffffffa0d8dd6000
  . GEN2  : 0xffffffa0d8dd6000 - 0xffffffa5a55d6000
  . GEN3  : 0xffffffa5a55d6000 - 0xffffffaa71dd6000
  . DATA  : 0xffffffaa71dd6000 - 0xffffffb0d7dd6000
  Metadata: 0xffffff909265d000 - 0xffffff90b265d000
  Bitmaps : 0xffffff90b265d000 - 0xffffff90b565d000
  Extra   : 0 - 0
```
</details>

<details>
<summary>Kernel trap at 0xffffff8010ee1d3f, type 14=page</summary>

```
panic(cpu 1 caller 0xffffff8010fb6113): Kernel trap at 0xffffff8010ee1d3f, type 14=page 
fault, registers:
CR0: 0x0000000080010033, CR2: 0xffffff802595fa30, CR3: 0x00000002797c81a7, CR4: 
0x00000000003626e0
RAX: 0xffffff708b0816b0, RBX: 0xffffff8011cd6d00, RCX: 0x0000000f9a8de380, RDX: 
0xffffff9a8de38800
RSP: 0xffffffc4b5a1bdf0, RBP: 0xffffffc4b5a1be10, RSI: 0xffffff9a8de38800, RDI: 
0xffffff8011cd6d00
R8:  0x0000000000000000, R9:  0x0000000000000000, R10: 0xffffff9a8cdc87f8, R11: 
0x000000015225c000
R12: 0xffffff9a8bea5100, R13: 0xffffffa0f3867290, R14: 0xffffff9a8de38800, R15: 
0x0000000000000100
RFL: 0x0000000000010202, RIP: 0xffffff8010ee1d3f, CS:  0x0000000000000008, SS:  
0x0000000000000010
Fault CR2: 0xffffff802595fa30, Error code: 0x0000000000000009, Fault CPU: 0x1, PL: 0, VF: 1

Panicked task 0xffffffaa8981c698: 12 threads: pid 749: chronod
Backtrace (CPU 1), panicked thread: 0xffffffa0f3866b30, Frame : Return Address
0xffffffc4b5a1b7d0 : 0xffffff8010e7207d 
0xffffffc4b5a1b820 : 0xffffff8010fc6164 
0xffffffc4b5a1b860 : 0xffffff8010fb5c77 
0xffffffc4b5a1b8b0 : 0xffffff8010e12951 
0xffffffc4b5a1b8d0 : 0xffffff8010e7235d 
0xffffffc4b5a1b9c0 : 0xffffff8010e71a07 
0xffffffc4b5a1ba20 : 0xffffff80115db40b 
0xffffffc4b5a1bb10 : 0xffffff8010fb6113 
0xffffffc4b5a1bc90 : 0xffffff8010fb5dfd 
0xffffffc4b5a1bce0 : 0xffffff8010e12951 
0xffffffc4b5a1bd00 : 0xffffff8010ee1d3f 
0xffffffc4b5a1be10 : 0xffffff8010f3a3a3 
0xffffffc4b5a1be40 : 0xffffff8010f395f8 
0xffffffc4b5a1be80 : 0xffffff8010f2528a 
0xffffffc4b5a1bee0 : 0xffffff8010f69b4c 
0xffffffc4b5a1bf10 : 0xffffff801132db05 
0xffffffc4b5a1bf40 : 0xffffff801145adc0 
0xffffffc4b5a1bfa0 : 0xffffff8010e12d96 

Process name corresponding to current thread (0xffffffa0f3866b30): chronod
Boot args: darkwake=0 ps2kbdonly=1

Mac OS version:
22G91

Kernel version:
Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; 
root:xnu-8796.141.3~6/RELEASE_X86_64
Kernel UUID: 8295C77B-2AAB-3471-81B5-438F59B1A36D
roots installed: 0
KernelCache slide: 0x0000000010a00000
KernelCache base:  0xffffff8010c00000
Kernel slide:      0x0000000010adc000
Kernel text base:  0xffffff8010cdc000
__HIB  text base: 0xffffff8010b00000
System model name: MacBookPro15,4 (Mac-53FDB3D8DB8CA971)
System shutdown begun: NO
Panic diags file available: YES (0x0)
Hibernation exit count: 0

System uptime in nanoseconds: 130514957180
Last Sleep:           absolute           base_tsc          base_nano
  Uptime  : 0x0000001e634c381b
  Sleep   : 0x0000000000000000 0x0000000000000000 0x0000000000000000
  Wake    : 0x0000000000000000 0x0000000554667049 0x0000000000000000
Compressor Info: 0% of compressed pages limit (OK) and 0% of segments limit (OK) with 0 
swapfiles and OK swap space
Zone info:
  Zone map: 0xffffff9a89e95000 - 0xffffffba89e95000
  . PGZ   : 0xffffff9a89e95000 - 0xffffff9a8be96000
  . VM    : 0xffffff9a8be96000 - 0xffffff9f58695000
  . RO    : 0xffffff9f58695000 - 0xffffffa0f1e95000
  . GEN0  : 0xffffffa0f1e95000 - 0xffffffa5be695000
  . GEN1  : 0xffffffa5be695000 - 0xffffffaa8ae95000
  . GEN2  : 0xffffffaa8ae95000 - 0xffffffaf57695000
  . GEN3  : 0xffffffaf57695000 - 0xffffffb423e95000
  . DATA  : 0xffffffb423e95000 - 0xffffffba89e95000
  Metadata: 0xffffff8025920000 - 0xffffff8045920000
  Bitmaps : 0xffffff8045920000 - 0xffffff8048920000
  Extra   : 0 - 0
```
</details>

<details>
<summary>panic(cpu 2 caller 0xffffff8008195d23): Possible memory corruption</summary>

```
panic(cpu 2 caller 0xffffff8008195d23): Possible memory corruption: 
pmap_pv_remove(0xffffff8008d2af00, 0xffffff801cbab000, 0x10da23, 
0x1000010da235d0, 0xffffffbaa09436d4, 0xffffffbaa0943670): pv not on hash, 
head: 0xffffff8b6b59fb80, 0x12e58c000, priors: 1 @pmap_internal.h:979
Panicked task 0xffffff99cda4f8b8: 11 threads: pid 630: diagnosticd
Backtrace (CPU 2), panicked thread: 0xffffff869ea550c8, Frame : Return 
Address
0xffffffbaa0943290 : 0xffffff800807207d 
0xffffffbaa09432e0 : 0xffffff80081c6164 
0xffffffbaa0943320 : 0xffffff80081b5c77 
0xffffffbaa0943370 : 0xffffff8008012951 
0xffffffbaa0943390 : 0xffffff800807235d 
0xffffffbaa0943480 : 0xffffff8008071a07 
0xffffffbaa09434e0 : 0xffffff80087db40b 
0xffffffbaa09435d0 : 0xffffff8008195d23 
0xffffffbaa0943700 : 0xffffff8008113b17 
0xffffffbaa0943780 : 0xffffff80081126a8 
0xffffffbaa0943800 : 0xffffff80081111be 
0xffffffbaa0943a80 : 0xffffff80081b5bb6 
0xffffffbaa0943af0 : 0xffffff8008012951 
0xffffffbaa0943b10 : 0xffffff8008011deb 
0xffffffbaa0943c70 : 0xffffff8008542a78 
0xffffffbaa0943cc0 : 0xffffff80085950de 
0xffffffbaa0943dd0 : 0xffffff8008569408 
0xffffffbaa0943e00 : 0xffffff8008563dd9 
0xffffffbaa0943e50 : 0xffffff80085641a7 
0xffffffbaa0943ec0 : 0xffffff800856437b 
0xffffffbaa0943f40 : 0xffffff800865adc0 
0xffffffbaa0943fa0 : 0xffffff8008012d96 

Process name corresponding to current thread (0xffffff869ea550c8): 
diagnosticd
Boot args: darkwake=0 ps2kbdonly=1

Mac OS version:
22G91

Kernel version:
Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; 
root:xnu-8796.141.3~6/RELEASE_X86_64
Kernel UUID: 8295C77B-2AAB-3471-81B5-438F59B1A36D
roots installed: 0
KernelCache slide: 0x0000000007c00000
KernelCache base:  0xffffff8007e00000
Kernel slide:      0x0000000007cdc000
Kernel text base:  0xffffff8007edc000
__HIB  text base: 0xffffff8007d00000
System model name: MacBookPro15,4 (Mac-53FDB3D8DB8CA971)
System shutdown begun: NO
Panic diags file available: YES (0x0)
Hibernation exit count: 0

System uptime in nanoseconds: 79036832252
Last Sleep:           absolute           base_tsc          base_nano
  Uptime  : 0x0000001266f65e77
  Sleep   : 0x0000000000000000 0x0000000000000000 0x0000000000000000
  Wake    : 0x0000000000000000 0x000000067f1ea9e5 0x0000000000000000
Compressor Info: 0% of compressed pages limit (OK) and 0% of segments 
limit (OK) with 0 swapfiles and OK swap space
Zone info:
  Zone map: 0xffffff8036090000 - 0xffffffa036090000
  . PGZ   : 0xffffff8036090000 - 0xffffff8038091000
  . VM    : 0xffffff8038091000 - 0xffffff8504890000
  . RO    : 0xffffff8504890000 - 0xffffff869e090000
  . GEN0  : 0xffffff869e090000 - 0xffffff8b6a890000
  . GEN1  : 0xffffff8b6a890000 - 0xffffff9037090000
  . GEN2  : 0xffffff9037090000 - 0xffffff9503890000
  . GEN3  : 0xffffff9503890000 - 0xffffff99d0090000
  . DATA  : 0xffffff99d0090000 - 0xffffffa036090000
  Metadata: 0xffffffba7d2e0000 - 0xffffffba9d2e0000
  Bitmaps : 0xffffffba9d2e0000 - 0xffffffbaa02e0000
  Extra   : 0 - 0
```
</details>

<details>
<summary>panic(cpu 4 caller 0xffffff8011595e03): Possible memory corruption</summary>

```
panic(cpu 4 caller 0xffffff8011595e03): Possible memory corruption: 
pmap_pv_remove(0xffffff801212af00,0xffffff8025fe0000,0x7ff0, 
0x10000007ff0000, 0xfffffff41f42f634, 0xfffffff41f42f5d0): null pv_list, 
priors: 1 @pmap_internal.h:903
Panicked task 0xffffffb431896e18: 27 threads: pid 585: SpringBoard
Backtrace (CPU 4), panicked thread: 0xffffffb433a680c8, Frame : Return 
Address
0xfffffff41f42f200 : 0xffffff801147207d 
0xfffffff41f42f250 : 0xffffff80115c6164 
0xfffffff41f42f290 : 0xffffff80115b5c77 
0xfffffff41f42f2e0 : 0xffffff8011412951 
0xfffffff41f42f300 : 0xffffff801147235d 
0xfffffff41f42f3f0 : 0xffffff8011471a07 
0xfffffff41f42f450 : 0xffffff8011bdb40b 
0xfffffff41f42f540 : 0xffffff8011595e03 
0xfffffff41f42f660 : 0xffffff8011513b17 
0xfffffff41f42f6e0 : 0xffffff80115115e6 
0xfffffff41f42f940 : 0xffffff80115b5bb6 
0xfffffff41f42f9b0 : 0xffffff8011412951 
0xfffffff41f42f9d0 : 0xffffff8011411deb 
0xfffffff41f42fb30 : 0xffffff801144e9eb 
0xfffffff41f42fd60 : 0xffffff801146327b 
0xfffffff41f42fdd0 : 0xffffff8011463927 
0xfffffff41f42fef0 : 0xffffff801159b533 
0xfffffff41f42ffa0 : 0xffffff8011412db6 

Process name corresponding to current thread (0xffffffb433a680c8): 
SpringBoard
Boot args: darkwake=0 ps2kbdonly=1 

Mac OS version:
22G91

Kernel version:
Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; 
root:xnu-8796.141.3~6/RELEASE_X86_64
Kernel UUID: 8295C77B-2AAB-3471-81B5-438F59B1A36D
roots installed: 0
KernelCache slide: 0x0000000011000000
KernelCache base:  0xffffff8011200000
Kernel slide:      0x00000000110dc000
Kernel text base:  0xffffff80112dc000
__HIB  text base: 0xffffff8011100000
System model name: MacBookPro15,4 (Mac-53FDB3D8DB8CA971)
System shutdown begun: NO
Panic diags file available: YES (0x0)
Hibernation exit count: 0

System uptime in nanoseconds: 62539098655
Last Sleep:           absolute           base_tsc          base_nano
  Uptime  : 0x0000000e8f9eee36
  Sleep   : 0x0000000000000000 0x0000000000000000 0x0000000000000000
  Wake    : 0x0000000000000000 0x000000068b636efe 0x0000000000000000
Compressor Info: 0% of compressed pages limit (OK) and 0% of segments 
limit (OK) with 0 swapfiles and OK swap space
Zone info:
  Zone map: 0xffffff9a9d726000 - 0xffffffba9d726000
  . PGZ   : 0xffffff9a9d726000 - 0xffffff9a9f727000
  . VM    : 0xffffff9a9f727000 - 0xffffff9f6bf26000
  . RO    : 0xffffff9f6bf26000 - 0xffffffa105726000
  . GEN0  : 0xffffffa105726000 - 0xffffffa5d1f26000
  . GEN1  : 0xffffffa5d1f26000 - 0xffffffaa9e726000
  . GEN2  : 0xffffffaa9e726000 - 0xffffffaf6af26000
  . GEN3  : 0xffffffaf6af26000 - 0xffffffb437726000
  . DATA  : 0xffffffb437726000 - 0xffffffba9d726000
  Metadata: 0xffffff9097196000 - 0xffffff90b7196000
  Bitmaps : 0xffffff90b7196000 - 0xffffff90ba196000
  Extra   : 0 - 0
```
</details>

<details>
<summary>panic(cpu 6 caller 0xffffff8006797925): Possible memory corruption</summary>

```
panic(cpu 6 caller 0xffffff8006797925): Possible memory corruption: 
pmap_pv_remove(0xffffffa42962bd80, 0x120430000, 0x2f08c6, 
0x6846002f08c610c, 0xfffffff3f77b3a04, 0xfffffe844130f180): pv not on 
hash, head: 0xffffffa42ad7ba00, 0x13315e000, priors: 1 
@pmap_internal.h:979
NULL bsd_info pointer
Panicked task 0xffffff95c4f708b8: 1 threads: unknown task
Backtrace (CPU 6), panicked thread: 0xffffff9a93fae598, Frame : Return 
Address
0xfffffff3f77b35f0 : 0xffffff800667207d 
0xfffffff3f77b3640 : 0xffffff80067c6164 
0xfffffff3f77b3680 : 0xffffff80067b5c77 
0xfffffff3f77b36d0 : 0xffffff8006612951 
0xfffffff3f77b36f0 : 0xffffff800667235d 
0xfffffff3f77b37e0 : 0xffffff8006671a07 
0xfffffff3f77b3840 : 0xffffff8006ddb40b 
0xfffffff3f77b3930 : 0xffffff8006797925 
0xfffffff3f77b3a30 : 0xffffff8006797c01 
0xfffffff3f77b3a80 : 0xffffff800672264e 
0xfffffff3f77b3bf0 : 0xffffff80067251b9 
0xfffffff3f77b3c50 : 0xffffff800672b7a4 
0xfffffff3f77b3ca0 : 0xffffff80066b09b5 
0xfffffff3f77b3ce0 : 0xffffff800665322e 
0xfffffff3f77b3d30 : 0xffffff80066b5c3e 
0xfffffff3f77b3d80 : 0xffffff80066c1f42 
0xfffffff3f77b3e50 : 0xffffff80066c6523 
0xfffffff3f77b3e80 : 0xffffff800666845e 
0xfffffff3f77b3eb0 : 0xffffff800661291d 
NULL bsd_info pointer

Process name corresponding to current thread (0xffffff9a93fae598): Unknown
Boot args: darkwake=0 ps2kbdonly=1

Mac OS version:
22G91

Kernel version:
Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; 
root:xnu-8796.141.3~6/RELEASE_X86_64
Kernel UUID: 8295C77B-2AAB-3471-81B5-438F59B1A36D
roots installed: 0
KernelCache slide: 0x0000000006200000
KernelCache base:  0xffffff8006400000
Kernel slide:      0x00000000062dc000
Kernel text base:  0xffffff80064dc000
__HIB  text base: 0xffffff8006300000
System model name: MacBookPro15,4 (Mac-53FDB3D8DB8CA971)
System shutdown begun: NO
Panic diags file available: YES (0x0)
Hibernation exit count: 0

System uptime in nanoseconds: 133169567550
Last Sleep:           absolute           base_tsc          base_nano
  Uptime  : 0x0000001f01865a25
  Sleep   : 0x0000000000000000 0x0000000000000000 0x0000000000000000
  Wake    : 0x0000000000000000 0x000000076ba8c2c8 0x0000000000000000
Compressor Info: 0% of compressed pages limit (OK) and 0% of segments 
limit (OK) with 0 swapfiles and OK swap space
Zone info:
  Zone map: 0xffffff8a91dd0000 - 0xffffffaa91dd0000
  . PGZ   : 0xffffff8a91dd0000 - 0xffffff8a93dd1000
  . VM    : 0xffffff8a93dd1000 - 0xffffff8f605d0000
  . RO    : 0xffffff8f605d0000 - 0xffffff90f9dd0000
  . GEN0  : 0xffffff90f9dd0000 - 0xffffff95c65d0000
  . GEN1  : 0xffffff95c65d0000 - 0xffffff9a92dd0000
  . GEN2  : 0xffffff9a92dd0000 - 0xffffff9f5f5d0000
  . GEN3  : 0xffffff9f5f5d0000 - 0xffffffa42bdd0000
  . DATA  : 0xffffffa42bdd0000 - 0xffffffaa91dd0000
  Metadata: 0xffffffffda883000 - 0xfffffffffa883000
  Bitmaps : 0xfffffffffa883000 - 0xfffffffffd883000
  Extra   : 0 - 0
```
</details>

**Attach your entire EFI folder**
[efi.zip](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/files/12568357/efi.zip)

**Attach system.log**
<details>
<summary>log</summary>

```
2023-09-10 18:36:46.599495+0500 0x77       Default     0x0                  0      0    kernel: [System Event] [2599493592] [ERROR] [Subsystem: launchd] [Event: data-protection] posix_spawn(): 86: Bad CPU type in executable
2023-09-10 18:36:50.031931+0500 0x2dd      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] FB0 Not waiting for VBL as path state is not active
2023-09-10 18:36:51.496308+0500 0x5f3      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [doSetPowerState] Returning immediately as current power state is same as new
2023-09-10 18:36:51.496354+0500 0x5f3      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [doSetPowerState] Returning immediately as current power state is same as new
2023-09-10 18:36:51.496378+0500 0x5f3      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [doSetPowerState] Returning immediately as current power state is same as new
2023-09-10 18:36:52.257429+0500 0x7a5      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] TxnHang1: FB0: IsTransactionComplete called following fakeVBL notification
2023-09-10 18:36:52.272904+0500 0x2dd      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] FB0: Flip called without enabling VBL
2023-09-10 18:36:52.324580+0500 0x137      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] FB0: VBlank Timeout Timer called in 51ms - fTransactionState = 0x0, fLiveState = 0x0 fOnline: 1
2023-09-10 18:36:52.324587+0500 0x137      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] Path state is 2
2023-09-10 18:36:52.369972+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.401326+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.434879+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.464783+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.496138+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.526580+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.557255+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.588658+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.618550+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.651995+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.683307+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.714574+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.745383+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.776427+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
2023-09-10 18:36:52.808822+0500 0x684      Default     0x0                  0      0    kernel: (AppleIntelKBLGraphicsFramebuffer) [IGFB][ERROR  ] [AGDC] Failed with status e00002bc for stream access aux
// AND SO ON
```
</details>

My installation process was:
- Created bootable MacOS usb as in OpenCore [instructions](https://dortania.github.io/OpenCore-Install-Guide/installer-guide/linux-install.html#making-the-installer).
- Installed MacOS
- Copied OC and BOOT folders in my EFI partition on second disk.

Thank you for you work and will also appreciate your help!

**Comment by stevezhengshiqi** [comment]
Thank you for the detailed description and log.
I can  see multiple panic on SpringBoard, configd, etc from the log. As for now, I don't see much abnormal logs. Only `[System Event] [2599493592] [ERROR] [Subsystem: launchd] [Event: data-protection] posix_spawn(): 86: Bad CPU type in executable` worries me as it prompts incompatible CPU type.

One thing worth trying is to dis-enable `BluetoothFixuo.kext` entry in config.plist, because it is known for incompatibilities on Sonoma.

**Comment by MrFoxPro** [comment]
> Thank you for the detailed description and log. I can see multiple panic on SpringBoard, configd, etc from the log. As for now, I don't see much abnormal logs. Only `[System Event] [2599493592] [ERROR] [Subsystem: launchd] [Event: data-protection] posix_spawn(): 86: Bad CPU type in executable` worries me as it prompts incompatible CPU type.
> 
> One thing worth trying is to dis-enable `BluetoothFixuo.kext` entry in config.plist, because it is known for incompatibilities on Sonoma.

CPU error could be because I trued to disable SIP. Not sure btw.
Yesturday I also tried Monterey and it hanged on unxiping 3.5GB archive.

I also ran memtest for a hour, had two passes, probably ram is fine.

I'm wondering if I can get more detailed logs `log show --predicate "processID == 0" --last boot --debug | grep "ERR"`, because right now I only can run it after reboot because of crash

**Comment by MrFoxPro** [comment]
Could it be because of running out of memory and macos trying to use swap?

**Comment by MrFoxPro** [comment]
Same errors here: https://www.tonymacx86.com/threads/sudden-macos-hang-ups-kernel-traps.297883/page-2
> Done. [Disabled the real EC](https://dortania.github.io/Getting-Started-With-ACPI/Universal/ec-fix.html#disabling-real-ec-desktops-only), [patched SMBus support](https://dortania.github.io/Getting-Started-With-ACPI/Universal/smbus.html).
I'm not sure, seems like this is already done in this repo. So I will run memtest for night and see if I have ram problems.

**Comment by MrFoxPro** [comment]
According to OC instructions, SSDT-MCHC should be changed to PC00.SMBC:

https://dortania.github.io/Getting-Started-With-ACPI/Universal/smbus-methods/manual.html#devicemanager
![IMG_20230912_015952](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/assets/17229619/08544837-9fec-45f2-b73e-425f1f845d98)

**Comment by stevezhengshiqi** [comment]
> According to OC instructions, SSDT-MCHC should be changed to PC00.SMBC:
> 
> https://dortania.github.io/Getting-Started-With-ACPI/Universal/smbus-methods/manual.html#devicemanager !

In the sample, they change the SMBUS' ACPI path because the ACPI path for their SMBus device is `PC00.SMBS`, shown in Windows Device Manager. Our laptop has traditional ACPI path for SMBus, so no need to change the path name.

I don't think patching SMBus will help with this issue. The `DVL0` device added by [SSDT-SBUS-MCHC.dsl](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/AcpiSamples/Source/SSDT-SBUS-MCHC.dsl) never exists on Apple MacBookPros. Full discussion can be found here: https://github.com/acidanthera/OpenCorePkg/pull/442

**Comment by MrFoxPro** [comment]
Hm. Is it safe to patch cfg unlock with my `XMAKB5R0P0603`?

**Comment by stevezhengshiqi** [comment]
@MrFoxPro You can try out the lastest 0E07 package in https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/main/BIOS/TM1701/XMAKB5R0P0E07.exe

**Comment by MrFoxPro** [comment]
I think it's memory leak. Whenever I start simulator and it fully loads, it takes almost full RAM, I have only ~100M free of 16GB. I'm not sure if it's correct behavior. 
![Screenshot 2023-09-12 at 10 15 08 AM](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/assets/17229619/c508bac7-ee22-4e11-976d-2b4cdc4391de)

**Comment by MrFoxPro** [comment]
As for now:
- I tried to compile custom sstd as I mentioned before with changing pci0 to pc00 and it didn't help.
- Tried several combinations of boot args enabling/disabling kexts
- Patched BIOS by unlocking CFG and set DMT to 64.
Nothing helped

I installed Sonoma laongside and WiFI driver doesn't work, but! Ethernet via type&c hub suddenly connected.

**Comment by stevezhengshiqi** [comment]
I don't think it's an EFI related issue, but rather a user space problem. If a specifix kext causes kernel panic, you can see clearly the kext name in the crash report.

Maybe related to SSD model. You can try NVMeFix or tune SetApfsTrimTimeout in config.plist.

**Comment by MrFoxPro** [comment]
Yep I will try

> I don't think it's an EFI related issue, but rather a user space problem. If a specifix kext causes kernel panic, you can see clearly the kext name in the crash report.
> 
> Maybe related to SSD model. You can try NVMeFix or tune SetApfsTrimTimeout in config.plist.
> 
> For Sonoma wifi, you need to make sure you install the latest beta. AirportItlwm_Sonoma used in v1.8.0 release only supports beta5+. Ref: [OpenIntelWireless/itlwm#883 (comment)](https://github.com/OpenIntelWireless/itlwm/issues/883#issuecomment-1670749680)

Yep thank you, I will try. Righ now I updated BIOS as you suggested and my MacOS now doesn't boot at all. I think I need to reinstall again.

Btw, can you suggest 100% working SSD model?

**Comment by MrFoxPro** [comment]
> After updating the BIOS, you need to patch cfg and 64dvmt again, or turn on the AppleXcpmCfgLock and reverse framebuffer-fbmem and framebuffer-stolenmem changes.
> 
> For compatible SSD model, I recommend referring to [dortania/bugtracker#192](https://github.com/dortania/bugtracker/issues/192)

Not quite understand, does ***A07 BIOS cobpatable with dvmt64 and cfg unlock patches?

**Comment by stevezhengshiqi** [comment]
Yes, it does compatible. However, upgrading BIOS will restore every changes you made to BIOS (e.g. cfg, 64dvmt) to factory setting, so you have to patch them again or use the original config.

**Comment by MrFoxPro** [comment]
So, I:
- Updated BIOS to latest, patched DVMT/CFG
- Installed NvmeFix
- Tried to change SMBIOS (unsucessful, didn't boot)

Same errors: kernel trap / memory corruptions
I will look forward to try more suitable SSD, I'm little confused because first time I installed on different disk and I saw similiar errors. But seems like they have both Silicon controllers, could this be a key?  
I have my ejected Samsung `PM981`, maybe I will try install there just out of curiosity.

Anyway, thank you for your work and especially for involvement in my issue!

**Comment by MrFoxPro** [comment]
Also, does system-boot or ssd slot matter? Or disk layout?

**Comment by MrFoxPro** [comment]
One of my guesses turned out to be true:
> Could it be because of running out of memory and macos trying to use swap?

I was lucky to buy used Samsung 960 EVO and now it's not crashing. Before, when I started apps, `swap` was always zero. Seems like macOS wasn't able to write to it and failed when tried to read from it. Now it's working. I set `SetApfsTimeout` to `0` as mentioned in SSD guide.

Thank you @stevezhengshiqi

**Comment by MrFoxPro** [comment]
I faced this issue again after a while. Seems like it happened because of running fstrim on linux. And seems like it could be fixed by removing/create new empty apfs partitions with some reserved size
